apps/common/models/i-casework.js

```diff
@@ -41,16 +40,19 @@ module.exports = class CaseworkModel extends Model {
   }
 
   save() {
-    const options = this.requestConfig({});
-    options.form = this.prepare();
-    options.method = 'POST';
-    if (!config.icasework.secret || !config.icasework.key && config.env !== 'production') {
-      return Promise.resolve({
-        createcaseresponse: {
-          caseid: 'mock caseid'
-        }
-      });
-    }
-    return this.request(options);
+    return Promise.resolve(this.prepare()).then(formData => {
```

Comment:
this.prepare is usually sync, however the Compose function/model above returns an async prepare step if needed to generate an auth token. This handles that by always resolving a promise whether it is sync or async.